src/v8/isolate_scope.rs

```diff
@@ -82,17 +141,54 @@ impl<'isolate> V8IsolateScope<'isolate> {
     pub(crate) fn new_dummy(isolate: &'isolate V8Isolate) -> V8IsolateScope<'isolate> {
         V8IsolateScope {
             isolate,
-            inner_handlers_scope: std::ptr::null_mut(),
-            inner_isolate_scope: std::ptr::null_mut(),
+            storage: V8IsolateScopeStorage::default(),
         }
     }
 
+    /// Returns a raw pointer to the isolate.
+    pub fn get_raw_isolate(&self) -> *mut v8_isolate {
+        self.isolate.get_raw()
+    }
+
+    /// Creates a new [V8IsolateScope] (or "restores it") with an
+    /// already existing storage for the scopes.
+    /// The storage provided must have been created precisely for the
+    /// isolate passed.
+    pub fn restore(isolate: &'isolate V8Isolate, storage: V8IsolateScopeStorage) -> Option<Self> {
```

Comment:
I wonder where do you need to use this function? the all idea of isolate scope is that its stack allocated object that guards the execution of a JS code. So in what case you need to restore and existing isolate scope?